src/Microsoft.AspNetCore.Razor/CodeGenerators/CSharpCodeGenerator.cs

```diff
@@ -82,12 +83,32 @@ public override CodeGeneratorResult Generate()
                             csharpCodeVisitor.Accept(Tree.Children);
                         }
                     }
+
+                    BuildAfterExecuteContent(writer, Tree.Children);
                 }
             }
 
             return new CodeGeneratorResult(writer.GenerateCode(), writer.LineMappingManager.Mappings);
         }
 
+        /// <summary>
+        /// Provides an entry point to append code (after execute content) to a generated Razor class.
+        /// </summary>
+        /// <param name="writer">The <see cref="CSharpCodeWriter"/> to receive the additional content.</param>
+        /// <param name="chunks">The list of <see cref="Chunk"/>s for the generated program.</param>
+        protected virtual void BuildAfterExecuteContent(CSharpCodeWriter writer, IList<Chunk> chunks)
```

Comment:
Decided offline to leave this as is b/c `Tree.Children` is an `IList<Chunk>` and creating another `List<Chunk>` isn't worth it.